### PR TITLE
raj/bad pv/drop_sys_with_only_nan

### PR DESCRIPTION
**Pull request by vrym2**
# Pull Request

## Description

This PR deals with the function that drops the systems which have only `NaN's` in a given single day. This PR incorporates #129 in its tests

Fixes #

## How Has This Been Tested?

The test has been performed on `passiv_datapipe1 and `pvoutput_datapipe`. It has also been tested with a constructed `xarray` where the `xarray` has three systems with one system only containing `NaN's`. After the test, the result `xarray` drops the system with only `NaN's`
- [ ] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ocf_datapipes/utils/utils.py [hunk 1]
```diff
@@ -19,6 +19,42 @@
 logger = logging.getLogger(__name__)
 
 
+def return_sys_indices_which_has_cont_nan(arr: np.ndarray, check_interval: int = 287) -> np.ndarray:
```
Comment (jacobbieker) [inline ocf_datapipes/utils/utils.py:22]:
```suggestion
def return_system_indices_which_has_contiguous_nan(arr: np.ndarray, check_interval: int = 287) -> np.ndarray:
```

Could the names of the functions be more explicit? Not sure if this should be contiguous or continuous, or something else, but just to keep it as explicit as possible

File: ocf_datapipes/utils/utils.py [hunk 2]
```diff
@@ -19,6 +19,42 @@
...
+    contigous 289 NaN's.
+
+    Args:
+        arr: Array of each system pvoutput values for a single day
+        check_interval: time range intervals respectively
+
+    """
+    # Checking the shape of the input array
+    # The array would be a 2d-array which consists of number of (time_utc, pv_system_id)
+    number_of_sys = arr.shape[1]
```
Comment (jacobbieker) [inline ocf_datapipes/utils/utils.py:35]:
```suggestion
    number_of_systems = arr.shape[1]
```

**Review by jacobbieker — CHANGES_REQUESTED** [review]
This looks good overall! Just a few changes to make a bit clearer, but looks good overall

File: ocf_datapipes.code-workspace [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
[the remark this answers is not in the document]
Comment (vrym2) [reply ocf_datapipes.code-workspace:1]: sure, I'll add this .gitignore!

File: ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+
```
[the remark this answers is not in the document]
Comment (vrym2) [reply ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py]: ok!

File: ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+from torchdata.datapipes.iter import IterDataPipe
+
+from ocf_datapipes.utils.utils import return_system_indices_which_has_contiguous_nan
+
+logger = logging.getLogger(__name__)
+
+
+@functional_datapipe("select_pv_systems_without_output")
+class DropPVSysWithOnlyNanInADayIterDataPipe(IterDataPipe):
+    """Remove any PV systems with less than 1 day of data"""
```
Comment (peterdudfield) [inline ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py:17]: need to update text

File: ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        # Reading the DataArray
+        for xr_dataset in self.source_datapipe:
+
+            dates_array = xr_dataset.coords["time_utc"].values
+
+            # Checking length of time series longer than standard intervals
+            if dates_array.size >= self.intervals:
+
+                # Collecting the time series data from 'time_utc' coordinate
+                for i in range(0, dates_array.size, self.intervals):
```
[the remark this answers is not in the document]
Comment (vrym2) [reply ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py:46]: I have added an extra test `test_execution_time`, and tested with constructed `xarray` with `date_range` of an year, using timeit package, the time taken to run through one iteration is 0.0009 seconds.

**Review by jacobbieker — APPROVED** [review]
Looks good! Just update that class name and I think it's good to go.

File: ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+from ocf_datapipes.utils.utils import return_system_indices_which_has_contiguous_nan
+
+logger = logging.getLogger(__name__)
+
+
+@functional_datapipe("select_pv_systems_without_output")
+class DropPVSysWithOnlyNanInADayIterDataPipe(IterDataPipe):
```
[the remark this answers is not in the document]
Comment (vrym2) [reply ocf_datapipes/select/drop_pv_sys_with_only_nan_in_a_day.py:16]: sure, I'll make the changes right away!

**Pull request merged**